pkg/p2p/libp2p/stream.go

```diff
@@ -0,0 +1,18 @@
+package libp2p
+
+import (
+	"github.com/libp2p/go-libp2p-core/helpers"
+	"github.com/libp2p/go-libp2p-core/network"
+)
+
+type Stream struct {
+	network.Stream
+}
+
+func (s *Stream) FullClose() error {
```

Comment:
This method provides important feature but is not tested. Its current implementation is trivial, but has no protection in tests from breaking changes.